frontend/src/MuiForms/DateSelector.tsx

```diff
@@ -0,0 +1,60 @@
+import 'dayjs/locale/en-gb'
+
+import { Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { DatePicker } from '@mui/x-date-pickers'
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
+import dayjs, { Dayjs } from 'dayjs'
+import { Fragment } from 'react'
+
+interface CustomTextInputProps {
+  label?: string
+  required?: boolean
+  disabled?: boolean
+  readOnly?: boolean
+  formContext?: any
+  value: string
+  onChange: (newValue: string) => void
+  InputProps?: any
+}
+
+export default function DateInput(props: CustomTextInputProps) {
+  const { onChange, value, label, formContext } = props
+
+  const theme = useTheme()
+  const handleChange = (dateInput: Dayjs | null) => {
+    if (dateInput && dateInput.isValid()) {
+      onChange(dateInput.toDate().toISOString().split('T')[0])
+    }
+    if (dateInput === null) {
+      onChange('')
+    }
+  }
+
+  const displayValue = (dateValue: string) => {
+    const [year, month, date] = dateValue.split('-')
+    return `${date}/${month}/${year}`
+  }
+
+  return (
+    <Fragment key={label}>
+      <Typography fontWeight='bold'>{label}</Typography>
+      {formContext.editMode && (
+        <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale='en-gb'>
```

Comment:
This can probably be far higher in the React tree, e.g. up at the top in `_app.tsx`.

Reply:
I thought the documentation meant you had to had to individually wrap the date pickers with these providers but if we can do it at the top level that'd be even better.

Reply:
On a different project we managed to just put it at the top level, so can probably do that here too unless they've changed something.